### DISCUSSION TITLE
How to import multiple images from a folder ?

### DISCUSSION THREAD

**Discussion by RebootGG** [vercel/next.js#34644] [Help]
Hi, 

I'd like to map over images in a `public` subfolder, so that my code doesn't have to look like this : 
```
  <Image src="/img-1.jpg" />
  <Image src="/img-2.jpg" />
  <Image src="/img-3.jpg" />
  <Image src="/img-4.jpg" />
  <Image src="/img-5.jpg" />
   ...
```

I've tried the following, but it doesn't work since it requires `context` from webpack
```
function importAll(r) {
  let images = {};
  r.keys().map((item, index) => { images[item.replace('./', '')] = r(item); });
  return images;
}

const images = importAll(require.context('./images', false, /\.(png|jpe?g|svg)$/));
```

Is there a way to do that with Next ?  Somehow I can't find anything on this issue.

Thanks for your help.

**Comment by harrisonratcliffe** [comment]
Did you find a solution for this?

**Comment by davidwwu** [comment]
+1
I'd love to utilize the power of `Image` and the convenient `blur` functionality but I have a bunch of images and was looking for a way to import all images from a folder.

**Reply by Ambushfall** [reply]
> +1 I'd love to utilize the power of `Image` and the convenient `blur` functionality but I have a bunch of images and was looking for a way to import all images from a folder.
I've spun up a working example to showcase that it's doable in NextJs without Issues.

[Gallery using Next 13 on Vercel](https://ambushfall-next13-experimental-app.vercel.app/gallery)
`next13: app/gallery/Gallery.tsx`
```ts
import { promises as fs } from 'fs'
import path from 'path'
import React from 'react';
import Image from 'next/image'
import styles from '../page.module.css'

const { container, main, title, grid, card } = styles

// using process.cwd() to get the path
// fs.readdir to get all files in directory
const GalleryPage = async () => {
    const imageDirectory = path.join(process.cwd(), '/public/midj');
    const imageFilenames = await fs.readdir(imageDirectory)
    console.log(imageFilenames)

    return (
        <div className={container}>
            <div className={main}>
                <Title Name='Gallery' Title={'Proof of Concept'} />
                <Gallery images={imageFilenames} />

            </div>
        </div>
    );
};

const Gallery = ({ images }: { images: Array<string>; }) => {
    return <div className={grid}>
        {images.map((el: string) => <Image className={card} width={150} height={150} alt={'alt'} src={`/midj/${el}`} key={el} />)}
    </div>
}

const Title = ({ Name, Title }: { Name: string, Title: String }) => {
    return (
        <h1 className={title}>
            {Name} {Title}
        </h1>
    )
}

export default GalleryPage;
```

check out the GitHub Repo that's public for the source

[Next 13 Experimental App - Testing Concepts](https://github.com/Ambushfall/experimentalapp)

**Reply by torn4dom4n** [reply]
This is so good.

**Reply by davidwwu** [reply]
Thanks for the response but I've already tried this. I got the error saying: `Error: The Serverless Function "api/gallery" is 107.93mb which exceeds the maximum size limit of 50mb. Learn More: https://vercel.link/serverless-function-size`. I ended up using the regular `<img>` tag and CSS to handle the loading -> showing animation.

**Reply by runoncedev** [reply]
Did you use the exact code as above? How many images are you trying to read?

**Reply by Michelangelo-Foresta** [reply]
It works, if you need help fixing something we can jump on a teams or something. It definitely works

**Reply by tushgaurav** [reply]
What if my images are not in the public directory, they are in the same directory as the current page.tsx file under images folder and i want to import a image based on url parameters, I want to do this because i wanted to utilize Nextjs image blur functionality on build time.
